packages/rollup-plugin-html/rollup-plugin-html.js

```diff
@@ -208,10 +209,63 @@ function rollupPluginHtml(pluginOptions) {
          * @param {OutputBundle} bundle
          */
         async generateBundle(options, bundle) {
-          generatedBundles.push({ name, options, bundle });
-          if (generatedBundles.length === outputCount) {
-            this.emitFile(await createHtmlAsset());
-          }
+          return new Promise(resolve => {
+            if (!options.dir) {
+              throw createError(`Output ${name} must have a dir option set.`);
+            }
+
+            generatedBundles.push({ name, options, bundle });
+
+            if (pluginOptions.outputBundleName) {
+              if (!multiOutputNames.includes(pluginOptions.outputBundleName)) {
+                throw createError(
+                  `outputName is set to ${pluginOptions.outputBundleName} but there was no ` +
+                    "output added with this name. Used .addOutput('name') to add it.",
+                );
+              }
+
+              if (pluginOptions.outputBundleName === name) {
+                // we need to emit the asset from this output's asset tree, but not all build outputs have
+                // finished building yet. create a function to be called later to emit the final output
+                // when all builds are finished
+                const { dir } = options;
+                deferredEmitHtmlFile = () =>
+                  createHtmlAsset(dir).then(asset => {
+                    this.emitFile(asset);
+                    resolve();
+                  });
+              }
+            }
+
+            if (generatedBundles.length === multiOutputNames.length) {
+              // this is the last build, emit the HTML file
+              if (deferredEmitHtmlFile) {
+                // emit to another build output's asset tree
+                deferredEmitHtmlFile().then(() => {
+                  resolve();
```

Comment:
Is this `resolve()` redundant?

Reply:
no, because the promise is not returned. this is to satisfy the linter to have a consistent return type.